### PR TITLE
Created playlist.py

### PR DESCRIPTION
**Pull request by daksh10110**
Added a file playlist.py to calculate the total length of a youtube playlist

not sure how to use it, I just added a python file to do the work, it can be imported into the main program where we need to get the duration of the playlist which I don't know.

Note: While running the program, youtube API must be written over there, which isn't included in this PR, neither should the API key be on Github.

Please let me know if there's more to do here. #3

### CODE REVIEW

**Comment by daksh10110** [comment]
Some help would be appreciated here, idk why is this PR failing tests.

**Comment by ashutoshkrris** [comment]
It's not your mistake @daksh10110 
I have missed some actions, that's why they are failing.

**Comment by daksh10110** [comment]
Oh, lemme know if there's some issue with the code.

**Comment by ashutoshkrris** [comment]
@daksh10110 please allow me some time to review your PR

File: core/playlist.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import os
+import re
+from datetime import timedelta
+from googleapiclient.discovery import build
+
+class Playlist:
+    def __init__(self, link_of_playlist):
+        self.api_key = os.environ.get("Enter API KEY Here") # Need to replace "Enter API KEY here" with YT API's key while running the program
+
```
Comment (ashutoshkrris) [inline core/playlist.py:9]: Missing self.youtube

File: core/playlist.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            else:
+                self.playlist_id = self.link_of_playlist[37::]
+        elif self.link_of_playlist[0:3] == "www":
+            self.playlist_id = self.link_of_playlist[30::]
+        else:
+            return "Not a valid playlist URL"
+            
+
+        while True:
+            self.pl_request = youtube.playlistItems.list(
```
Comment (ashutoshkrris) [inline core/playlist.py:39]: youtube to be replaced with self.youtube

File: core/playlist.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                pageToken = self.next_page_token
+            )
+
+            self.pl_response = self.pl_request.execute()
+
+            self.vid_ids = []
+            for item in self.pl_response['items']:
+                self.vid_ids.append(item['contentDetails']['videoId'])
+
+            self.vid_request = youtube.videos().list(
```
Comment (ashutoshkrris) [inline core/playlist.py:52]: youtube to be replaced with self.youtube

File: core/playlist.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                self.playlist_id = self.link_of_playlist[37::]
+        elif self.link_of_playlist[0:3] == "www":
+            self.playlist_id = self.link_of_playlist[30::]
+        else:
+            return "Not a valid playlist URL"
+            
+
+        while True:
+            self.pl_request = youtube.playlistItems.list(
+                part = "contentdeetails",
```
Comment (ashutoshkrris) [inline core/playlist.py:40]: Typo in contentDetails

File: core/playlist.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            )
+
+            self.pl_response = self.pl_request.execute()
+
+            self.vid_ids = []
+            for item in self.pl_response['items']:
+                self.vid_ids.append(item['contentDetails']['videoId'])
+
+            self.vid_request = youtube.videos().list(
+                part="contentdetails",
```
Comment (ashutoshkrris) [inline core/playlist.py:53]: Typo in contentDetails

**Review by ashutoshkrris — CHANGES_REQUESTED** [review]
Hey @daksh10110 
I have highlighted few mistakes, please correct them

**Comment by daksh10110** [comment]
@ashutoshkrris I have made the necessary changes, is it ok now?

**Review by ashutoshkrris — CHANGES_REQUESTED** [review]
@daksh10110  It seems, self.youtube is still not initialized anywhere

**Comment by daksh10110** [comment]
My bad, added it now.

**Comment by daksh10110** [comment]
My bad, forgot to initialize it earlier.

File: core/playlist.py [hunk 6]
```diff
@@ -21,6 +21,7 @@ def __init__(self, link_of_playlist):
         self.pl_response = None
         
         self.vid_ids = None
+        self.youtube = None
```
Comment (ashutoshkrris) [inline core/playlist.py]: self.youtube must not be initialized with None as it is the one that uses build method of the googleapiclient.discovery module. 
It should be:
`self.youtube = build('youtube', 'v3', developerKey=self.api_key)`

**Comment by ashutoshkrris** [comment]
Great going, @daksh10110 
If possible, could you please provide some examples of how can I calculate the duration of speeds other than 1?

**Review by ashutoshkrris — APPROVED** [review]

**Pull request merged**